sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusSessionsEndToEndTests.cs

```
@@ -729,8 +737,14 @@ public class DrainModeTestJobTopic
                 _drainValidationPreDelay.Set();
                 await DrainModeHelper.WaitForCancellationAsync(cancellationToken);
                 Assert.True(cancellationToken.IsCancellationRequested);
-                await messageSession.CompleteMessageAsync(msg);
-                _drainValidationPostDelay.Set();
+                try
+                {
+                    await messageSession.CompleteMessageAsync(msg);
+                }
+                finally
+                {
+                    _drainValidationPostDelay.Set();
```

This should allow failed tests to fail faster and have better diagnosability.